openhands/integrations/gitlab/gitlab_service.py

```diff
@@ -523,6 +530,146 @@ async def create_mr(
 
         return response['web_url']
 
+    def _determine_microagents_path(self, repository_name: str) -> str:
+        """Determine the microagents directory path based on repository name."""
+        actual_repo_name = repository_name.split('/')[-1]
+
+        if actual_repo_name == 'openhands-config':
+            # For GitLab with repository name "openhands-config", scan "microagents" folder
+            return 'microagents'
+        else:
+            # Default behavior: look for .openhands/microagents directory
+            return '.openhands/microagents'
```

Comment:
It seems like this method is shared across all the service classes (with the exception of some logic being modified for gitlab)

Let's move it into `BaseGitService` class so that its inherited by the service classes - you can override the implementation for gitlab specifically

Same applies for the `_create_microagent_response` method